pkg/cas/content_addressable_storage_server_test.go

```diff
@@ -0,0 +1,88 @@
+package cas_test
+
+import (
+	"context"
+	"testing"
+
+	remoteexecution "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/buildbarn/bb-storage/internal/mock"
+	"github.com/buildbarn/bb-storage/pkg/blobstore/buffer"
+	"github.com/buildbarn/bb-storage/pkg/cas"
+	"github.com/buildbarn/bb-storage/pkg/digest"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestContentAddressableStorageServerBatchReadBlobsSuccess(t *testing.T) {
+	ctrl, ctx := gomock.WithContext(context.Background(), t)
+	defer ctrl.Finish()
+
+	digest1 := digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000001", 123)
+	digest2 := digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000002", 234)
+
+	request := &remoteexecution.BatchReadBlobsRequest{
+		Digests: []*remoteexecution.Digest{
+			digest1.GetPartialDigest(),
+			digest2.GetPartialDigest(),
+		},
+		InstanceName: "ubuntu1804",
+	}
+
+	contentAddressableStorage := mock.NewMockBlobAccess(ctrl)
+
+	a := make([]byte, 123)
+	buf := buffer.NewValidatedBufferFromByteSlice(a)
+	contentAddressableStorage.EXPECT().Get(ctx, digest1).Return(buf)
+	b := make([]byte, 234)
+	buf2 := buffer.NewValidatedBufferFromByteSlice(b)
+	contentAddressableStorage.EXPECT().Get(ctx, digest2).Return(buf2)
+
+	contentAddressableStorageServer := cas.NewContentAddressableStorageServer(contentAddressableStorage, 1<<16)
+
+	response, err := contentAddressableStorageServer.BatchReadBlobs(ctx, request)
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+	require.Equal(t, &remoteexecution.BatchReadBlobsResponse{
+		Responses: []*remoteexecution.BatchReadBlobsResponse_Response{
+			{
+				Digest: digest1.GetPartialDigest(),
+				Data:   a,
+				Status: status.Convert(nil).Proto(),
```

Comment:
Nit: I think these `Status:` lines can be removed entirely, right? This evaluates to nil.

Reply:
When I tried without the `Status:` lines the test failed and showed a diff where the expected output had no `status` field but the actual output had `status: <>`.

Reply:
Ah, interesting. I incorrectly thought that `status.Convert(nil).Proto()` would evaluate to `nil`. I think that's even more of a reason that we shouldn't use this construct, as it makes it harder to understand what's going on.

Now the question becomes: what do we want to return in this case? I suspect that clients will generally support both. Returning `nil` results in a smaller serialized representation. But an empty Status object (the existing practice) is also acceptable.

If we do want to return an empty Status object, I guess we could improve readability by using this:

```go
import (
    status_pb "google.golang.org/genproto/googleapis/rpc/status"
)

    {
        Status: &status_pb.Status{},
    }
```